meshtastic/interfaces/ble/lifecycle_service.py

```
@@ -0,0 +1,1303 @@
+        # Guard against sentinel "unknown" address; prefer previous active
+        # address because callback metadata can be stale.
+        address_for_registry = (
+            getattr(previous_client, "address", None)
+            if previous_client is not None
+            else (address if address != "unknown" else iface.address)
+        )
+        addr_disconnect_key = _addr_key(address_for_registry)
+        return iface._sorted_address_keys(addr_disconnect_key, alias_key), should_schedule_reconnect
+
+    @staticmethod
+    def _resolve_disconnect_target(
```

Ruff (0.15.5)

[warning] Too many return statements (7 > 6)

(PLR0911)